docs/en/user_guides/3_inference.md

```diff
@@ -4,22 +4,130 @@ MMSegmentation provides pre-trained models for semantic segmentation in [Model Z
+
+The attributes in `SegDataSample` are divided into several parts:
+
+- `gt_sem_seg` (PixelData) - Ground truth of semantic segmentation.
+- `pred_sem_seg` (PixelData) - Prediction of semantic segmentation.
+- `seg_logits` (PixelData) - Predicted logits of semantic segmentation.
+
+Example:
 
 ```python
 from mmseg.apis import init_model, inference_model
 from mmseg.utils import register_all_modules
-# Specify the path to model config and checkpoint file
-config_file = 'configs/pspnet/pspnet_r50-d8_512x1024_80k_cityscapes.py'
-checkpoint_file = 'checkpoints/pspnet_r50-d8_512x1024_80k_cityscapes_20200606_112131-2376f12b.pth'
+
+config_path = 'configs/pspnet/pspnet_r50-d8_4xb2-40k_cityscapes-512x1024.py'
+checkpoint_path = 'checkpoints/pspnet_r50-d8_512x1024_40k_cityscapes_20200605_003338-2966598c.pth'
+img_path = 'demo/demo.png'
+
 # register all modules in mmseg into the registries
 register_all_modules()
+
+model = init_model(config_path, checkpoint_path)
+result = inference_model(model, img_path)
+```
+
+### mmseg.apis.show_result_pyplot
+
+Visualize the segmentation results on the image.
+
+Parameters:
+
+- model (nn.Module) - The loaded segmentor.
+- img (str or np.ndarray) - Image filename or loaded image.
+- result (SegDataSample) - The prediction SegDataSample result.
+- opacity (float) - Opacity of painted segmentation map. Default `0.5`, must be in `(0, 1]` range.
+- title (str) - The title of pyplot figure. Default is ''.
+- draw_gt (bool) - Whether to draw GT SegDataSample. Default to `True`.
+- draw_pred (draws_pred) - Whether to draw Prediction SegDataSample. Default to `True`.
+- wait_time (float) - The interval of show. Default to `0`.
+- show (bool) - Whether to display the drawn image. Default to `True`.
+- save_dir (str, optional) - Save file dir for all storage backends. If it is `None`, the backend storage will not save any data.
+- out_file (str, optional) - Path to output file. Default to `None`.
+
+Example:
+
+```python
+from mmseg.apis import init_model, inference_model, show_result_pyplot
+from mmseg.utils import register_all_modules
+
+config_path = 'configs/pspnet/pspnet_r50-d8_4xb2-40k_cityscapes-512x1024.py'
+checkpoint_path = 'checkpoints/pspnet_r50-d8_512x1024_40k_cityscapes_20200605_003338-2966598c.pth'
+img_path = 'demo/demo.png'
+
+# register all modules in mmseg into the registries
+register_all_modules()
+
 # build the model from a config file and a checkpoint file
 model = init_model(config_file, checkpoint_file, device='cuda:0')
-# test image pair, and save the results
-img = 'demo/demo.png'
-result = inference_model(model, img)
+
+# inference on given image
+result = inference_model(model, img_path)
+
+# visualize the segmentation result
+show_result_pyplot(model, img_path, result)
```

Comment:
I think we should add `wait_time` in example, as the default value of it is 0, and users might have question why they don't see anything after run the example.